google/resource_compute_global_forwarding_rule_test.go

```diff
@@ -254,6 +321,130 @@ func testAccComputeGlobalForwardingRule_basic1(fr, proxy1, proxy2, backend, hc,
 	}`, fr, proxy1, proxy2, backend, hc, urlmap)
 }
 
+func testAccComputeGlobalForwardingRule_labels(fr, proxy1, proxy2, backend, hc, urlmap string) string {
+	return fmt.Sprintf(`
+	resource "google_compute_global_forwarding_rule" "foobar" {
+		description = "Resource created for Terraform acceptance testing"
+		ip_protocol = "TCP"
+		name = "%s"
+		port_range = "80"
+		target = "${google_compute_target_http_proxy.foobar1.self_link}"
```

Comment:
I believe target is optional. Since you are already testing the "setting a target" part in the basic tests, you can speed up this test by removing the target. You can then remove the target http proxies, health checks and backend service.

Reply:
I got so excited when I read this, but an attempt yields:

```
		Errors: []string{"google_compute_global_forwarding_rule.foobar: \"target\": required field is not set"}
```
😞

Reply:
Oh. I was reading the docs for `compute_forwarding_rule` for which the target field is optional... https://www.terraform.io/docs/providers/google/r/compute_forwarding_rule.html

For `global_forwarding_rule`, we set the attribute as `Required=true`. Is it an artificial constraint we added or is it constrained by the API. It seems weird to me that the behavior of the API for the global vs regional forwarding rule would differ on this.

Reply:
I just tested, GCP API requires target to be set for global forwarding rule. You can forget this thread.